homeassistant/components/mealie/todo.py

```diff
@@ -0,0 +1,260 @@
+        """Delete items from the list."""
+        try:
+            for uid in uids:
+                await self.coordinator.client.delete_shopping_item(uid)
+        except MealieError as exception:
+            raise HomeAssistantError(
+                f"An error occurred deleting item(s) on {self._shopping_list.name}"
+            ) from exception
+
+        await self.coordinator.async_refresh()
+
+    async def async_move_todo_item(
+        self, uid: str, previous_uid: str | None = None
+    ) -> None:
+        """Re-order an item on the list."""
+        if uid == previous_uid:
+            return
+        list_items: list[ShoppingItem] = self.coordinator.shopping_list_items[
+            self._shopping_list.list_id
+        ]
+
+        item_idx = {itm.item_id: idx for idx, itm in enumerate(list_items)}
+        if uid not in item_idx:
+            raise HomeAssistantError(f"Item '{uid}' not found in shopping list")
+        if previous_uid and previous_uid not in item_idx:
+            raise HomeAssistantError(
+                f"Item '{previous_uid}' not found in shopping list"
+            )
+        dst_idx = item_idx[previous_uid] + 1 if previous_uid else 0
+        src_idx = item_idx[uid]
+        src_item = list_items.pop(src_idx)
+        if dst_idx > src_idx:
+            dst_idx -= 1
+        list_items.insert(dst_idx, src_item)
+
+        for position, item in enumerate(list_items):
+            mutate_shopping_item = MutateShoppingItem()
+            mutate_shopping_item.list_id = item.list_id
+            mutate_shopping_item.item_id = item.item_id
+            mutate_shopping_item.position = position
+            mutate_shopping_item.is_food = item.is_food
+            mutate_shopping_item.quantity = item.quantity
+            mutate_shopping_item.label_id = item.label_id
+            mutate_shopping_item.note = item.note
+            mutate_shopping_item.checked = item.checked
+
+            if item.is_food:
+                mutate_shopping_item.food_id = item.food_id
+                mutate_shopping_item.unit_id = item.unit_id
+
+            await self.coordinator.client.update_shopping_item(
+                mutate_shopping_item.item_id, mutate_shopping_item
+            )
+
+        await self.coordinator.async_refresh()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+
+        self._async_update_attrs()
+        super()._handle_coordinator_update()
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Update todo attributes."""
+
+        items = []
+
+        if self._shopping_list.list_id in self.coordinator.shopping_list_items:
+            for item in self.coordinator.shopping_list_items[
+                self._shopping_list.list_id
+            ]:
+                todo_item = _convert_api_item(item)
+                items.append(todo_item)
+
+        self._attr_todo_items = items
```

Comment:
I think this all can be removed now

Reply:
The _handle_coordinator_update can yes, still need the _async_update_attrs

Reply:
No, because `_async_update_attrs` is updating `self._attr_todo_items`, but you already have a `@property` for `todo_items` so setting `self._attr_todo_items` doesn't have any effect